exonum/src/node/state.rs

```diff
@@ -870,6 +870,19 @@ impl State {
             }
         }
 
+        // Sort proposes, so the order of processing is predictable.
+        full_proposes.sort_unstable_by(|a, b| {
```

Comment:
I'm not sure I understand what sorting achieves here. Is it useful for testing purposes only?

Reply:
Not sure. It guarantees that proposes are processed according to the round number (primary key), and only then according to the hash.
The second one is useful for testing, yeah, but the first one is useful overall IMO - why would we want to process rounds in order 3-0-2-1?